File: Leet/7.reverse_integer-翻转整数/7.reverse_integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    let res = 0;
    while (x != 0) {
        let tmp = res * 10 + x % 10;
        if (Math.floor(tmp / 10) != res) {
            return 0;
        }
        res = tmp;
        x = Math.floor(x / 10);
    }
    return res;
};

File: Algorithms/7.reverse_integer-翻转整数/7.reverse_integer.java
class Solution {
    public int reverse(int x) {
        int res = 0;
        while (x != 0) {
            int tmp = res * 10 + x % 10;
            if (tmp / 10 != res) {
                return 0; // 溢出
            }
            res = tmp;
            x = x / 10;
        }
        return res;
    }
}